### PR TITLE
Implement ItemFishedEvent

### PR DESCRIPTION
**Pull request by famous1622**
Another event required by CoFH Core.

### CODE REVIEW

File: patchwork-events-entity/src/main/java/net/minecraftforge/event/entity/player/ItemFishedEvent.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import net.minecraft.entity.projectile.FishingBobberEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.DefaultedList;
+
+/**
+ * This event is called when a player fishes an item.
+ *
+ * <p>This event is cancellable.
+ * Canceling the event will cause the player to receive no items at all.
+ * The hook will still take the damage specified
```
Comment (kitlith) [inline patchwork-events-entity/src/main/java/net/minecraftforge/event/entity/player/ItemFishedEvent.java:37]:
```suggestion
 * <p>This event is cancellable.
 * Canceling the event will cause the player to receive no items at all.
 * The hook will still take the damage specified</p>
```

**Comment by kitlith** [comment]
Other than that single nit, looks solid.

File: patchwork-events-entity/src/main/java/net/minecraftforge/event/entity/player/ItemFishedEvent.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+ */
+
+public class ItemFishedEvent extends PlayerEvent {
```
Comment (TheGlitch76) [inline patchwork-events-entity/src/main/java/net/minecraftforge/event/entity/player/ItemFishedEvent.java:40]: This is cancellable in forge but not here

File: patchwork-events-entity/src/main/java/net/patchworkmc/mixin/event/entity/MixinFishingBobberEntity.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+		if (event.isCanceled()) {
+			this.remove();
+			cir.setReturnValue(event.getRodDamage());
+		}
+	}
+
+	@Inject(method = "method_6957", at = @At("RETURN"), cancellable = true)
+	private void onRodDamage(ItemStack itemStack, CallbackInfoReturnable<Integer> cir) {
+		Integer damage = rodDamage.get();
```
Comment (TheGlitch76) [inline patchwork-events-entity/src/main/java/net/patchworkmc/mixin/event/entity/MixinFishingBobberEntity.java:63]: This is always -1

File: patchwork-events-entity/src/main/java/net/patchworkmc/mixin/event/entity/MixinFishingBobberEntity.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+@Mixin(FishingBobberEntity.class)
+public abstract class MixinFishingBobberEntity {
+	@Shadow
+	private boolean field_7176; // is stuck to block
+
+	@Shadow
+	public abstract void remove();
+
+	@Unique
+	private final ThreadLocal<Integer> rodDamage = ThreadLocal.withInitial(() -> -1);
```
Comment (TheGlitch76) [inline patchwork-events-entity/src/main/java/net/patchworkmc/mixin/event/entity/MixinFishingBobberEntity.java:48]: This is never reassigned?

**Pull request merged**